backend/routes/userSession.js

```diff
@@ -0,0 +1,80 @@
+const express = require("express");
+const session = require('express-session');
+const {PrismaSessionStore} = require('@quixo3/prisma-session-store')
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+const userSessionRouter = express.Router();
+
+const userSession = (session({
+    secret: process.env.SESSION_SECRET,
+    saveUninitialized: false,
+    resave: false,
+    cookie: {
+        secure: false,
+        httpOnly: true,
+        maxAge: 1000 * 60 * 30
+    },
+    store: new PrismaSessionStore(
+        prisma,
+        {
+            checkPeriod: 2 * 60 * 1000,  //ms
+            dbRecordIdIsSessionId: true,
+            dbRecordIdFunction: undefined,
+        },
+    )
+}));
+
+const authenticateSession = ( async (req, res, next) => {
```

Comment:
this function can be cleaned up there is a ton of nested trys and conditional statements. See if you can simplify this logic and make it more readable